### PR TITLE
Multiple rustdoc fixes

### LINKED ISSUE #48733: rustdoc -- sidebar doesn't work in function documentation
Sidebar is entirely empty and getting an error in console:

TypeError: sidebar is undefined [main.js:1497:13](https://github.com/rust-lang/rust/blob/1e1bfc715f8dfe36a38d8fae36fcbe92e7463a3f/src/librustdoc/html/static/main.js#L1497)

ss

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #48733.

r? @QuietMisdreavus

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -1776,22 +1776,33 @@
...
             return;
         }
-        if (hasClass(next, 'docblock') ||
-            (hasClass(next, 'stability') &&
-             hasClass(next.nextElementSibling, 'docblock'))) {
-            insertAfter(toggle.cloneNode(true), e.childNodes[e.childNodes.length - 1]);
-        }
-        if (hasClass(e, 'impl')) {
+        if (checkIfThereAreMethods(next.childNodes) &&
+            (hasClass(next, 'docblock') ||
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/static/main.js]: Is this the right way to combine these conditions? Won't this also bar regular docblocks (the original ones, with hand-written docs) from getting fold toggles?

**Comment by onur** [comment]
@GuillaumeGomez I've been wrong, switching position of main.js and search-index.js doesn't fix #48733

**Comment by GuillaumeGomez** [comment]
So I remove it.

**Comment by QuietMisdreavus** [comment]
This PR has removed the per-method fold-toggle buttons on inherent impl blocks.

image

**Comment by QuietMisdreavus** [comment]
```
[01:13:02] ---- [rustdoc] rustdoc/fn-sidebar.rs stdout ----
[01:13:02] 	
[01:13:02] error: htmldocck failed!
[01:13:02] status: exit code: 1
[01:13:02] command: "/usr/bin/python2.7" "/checkout/src/etc/htmldocck.py" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc/fn-sidebar.stage2-x86_64-unknown-linux-gnu" "/checkout/src/test/rustdoc/fn-sidebar.rs"
[01:13:02] stdout:
[01:13:02] ------------------------------------------
[01:13:02] 
[01:13:02] ------------------------------------------
[01:13:02] stderr:
[01:13:02] ------------------------------------------
[01:13:02] 17: @has check failed
[01:13:02] 	File does not exist 'foo/constant.bar.html'
[01:13:02] 	// @has foo/constant.bar.html
[01:13:02] 18: @has check failed
[01:13:02] 	File does not exist 'foo/constant.bar.html'
[01:13:02] 	// @has - '//*[@class="sidebar-elems"]' ''
[01:13:02] 
[01:13:02] Encountered 2 errors
[01:13:02] 
[01:13:02] ------------------------------------------
[01:13:02] 
[01:13:02] thread '[rustdoc] rustdoc/fn-sidebar.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:2893:9
[01:13:02] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[01:13:02] 
[01:13:02] 
[01:13:02] failures:
[01:13:02]     [rustdoc] rustdoc/fn-sidebar.rs
```

**Pull request merged**